library/core/src/cell/lazy.rs

```diff
@@ -36,8 +34,8 @@ enum State<T, F> {
 /// //   92
 /// //   92
 /// ```
-#[unstable(feature = "lazy_cell", issue = "109736")]
-pub struct LazyCell<T, F = fn() -> T> {
+#[stable(feature = "lazy_cell_fn_pointer", since = "CURRENT_RUSTC_VERSION")]
+pub struct LazyCell<T, #[unstable(feature = "lazy_cell", issue = "109736")] F = fn() -> T> {
```

Comment:
I think it could be better to do the opposite - stabilize the current feature as `lazy_cell` so any `#![feature(lazy_cell)]` uses get a correct warning (and since it is the core functionality), then gate the F parameter under something like `lazy_cell_fn_parameter`

Reply:
I did it this way because [it's what the dev guide says to do](https://std-dev-guide.rust-lang.org/development/stabilization.html#partial-stabilizations) - or at least it's [what was done in the example PR](https://github.com/rust-lang/rust/pull/94640/files).

Reply:
Oh - I was asked to do the opposite when partially stabilizing this, which is why we have a gate `lazy_cell` in the first place 😆 https://github.com/rust-lang/rust/pull/105587#issuecomment-1489184632

Reply:
Honestly, I think that's a better policy, but I'll wait for further feedback from the team before changing anything.